### PR TITLE
Feature/account registration

### PR DESCRIPTION
**Pull request by BartoliniAndBorderCollies**
### Description
User of the application is now able to write a command and create account in the database.

### Motivation
This is the basic functionality of the application.

### Changes introduced by this PR

- Corrections of commits from the previous PR
- improvement of structure of the application

Furthermore, this branch makes the whole functionality to create an account for the user by:

- creating a run() in `Main`
- creating `MenuController` which interacts with the user to listen the command
- reacting for user input by sending the request through command class, controller class, service class to the repository class
- taking the necessary information from the user in order to create an account in the db

### Testing
Application connects and persist data in DB.

### CODE REVIEW

File: src/main/java/org/klodnicki/Main.java [hunk 1]
```diff
@@ -1,18 +1,13 @@
...
-        // Check database version (because there is nothing else in database)
-        String sql = "select version()";
-
-        String result = session.createNativeQuery(sql, String.class).getSingleResult();
-        System.out.println(result);
-
-        session.close();
-        HibernateUtil.shutdown();
+        do {
+            MenuController menuController = new MenuController();
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/Main.java:8]: Do not create new object of MenuController inside of a loop.

File: src/main/java/org/klodnicki/Main.java [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
+            menuController.run();
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/Main.java:9]: run() shouldn't be in loop. It's meant to be run only once.

File: src/main/java/org/klodnicki/controller/AccountController.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            menuController.displayOnMenu(CREATE_INFO);
+            String firstName = menuController.displayOnMenuAndAskForInput(FIRST_NAME);
+            String secondName = menuController.displayOnMenuAndAskForInput(SECOND_NAME);
+            String lastName = menuController.displayOnMenuAndAskForInput(LAST_NAME);
+            String pesel = menuController.displayOnMenuAndAskForInput(PESEL);
+            String phoneNumber = menuController.displayOnMenuAndAskForInput(PHONE_NUMBER);
+            String email = menuController.displayOnMenuAndAskForInput(EMAIL);
+            String address = menuController.displayOnMenuAndAskForInput(ADDRESS);
+
+            accountService.create(firstName, secondName, lastName, pesel, phoneNumber, email, address);
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/controller/AccountController.java:37]: Try block is unnecessarily so large. It should only contain create method which throws your `IllegalArgumentException`.

File: src/main/java/org/klodnicki/controller/AccountController.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+        } catch (IllegalArgumentException e) {
+            menuController.displayOnMenu(e.getMessage());
+        }
+    }
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/controller/AccountController.java:44]: The method ends without ever giving user information about successfully creating account.

File: src/main/java/org/klodnicki/controller/MenuController.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    public void run() {
+        menuService.addMainMenuCommands(this);
+        menuView.update(WELCOME);
+
+        String userInput = askForInput(USER_COMMAND);
+        menuService.executeCommand(userInput);
+
+        // wynik operacji wyżej
+        menuView.update("");
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/controller/MenuController.java]: Why is there view updated with empty String?

File: src/main/java/org/klodnicki/entity/Account.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import jakarta.persistence.*;
+
+@Entity
+@Table(name = "accounts")
+public class Account {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", nullable = false)
+    private Long id;
+    @Column(nullable = false)
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/entity/Account.java]: Add property `name` to the `@Column` to follow database convention of naming columns.
i.e. `@Column(name = "first_name")`

File: src/main/java/org/klodnicki/service/MenuService.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+public class MenuService {
+
+    private final Menu menu = new Menu();
+
+    public void addMainMenuCommands(MenuController menuController) { // tu jest taki parametr bo AccountController go potrzebuje
+        menu.addCommand(new CreateAccount(new AccountController(menuController)));
+    }
+
+    public void executeCommand(String userInput) {
+        // TODO: walidacja userInput
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/service/MenuService.java]: Is this TODO still to be done?

File: src/main/java/org/klodnicki/model/Menu.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.List;
+
+public class Menu {
+    private final List<MenuCommand> commands = new ArrayList<>();;
+
+    public void addCommand(MenuCommand command) {
+        commands.add(command);
+    }
+
+    public List<MenuCommand> getCommands() {
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/model/Menu.java:16]: is this method used?

**Review by JasinskiSz — CHANGES_REQUESTED** [review]
I left some comments requesting changes. The first biggest issue is with do while loop in main where inside is a run method. It doesn't make sense as you'll even print multiple "Welcome to the Library Manager app!" messages which are meant to be showed once on the app start for user.

Take a look at the places where I asked for changes and reply to them if something is unclear to you. Thanks!

**Comment by JasinskiSz** [comment]
Also I forgot. I didn't checked it in my IntelliJ. I'll do that in the evening, after 7 PM.

**Comment by BartoliniAndBorderCollies** [comment]
OK thanks for your comments. changes done. Please check.

File: src/main/java/org/klodnicki/model/Menu.java (hunk 8 above)
Comment (JasinskiSz) [reply src/main/java/org/klodnicki/model/Menu.java:16]: It is now used in `MenuService`. Great job!

File: src/main/java/org/klodnicki/controller/MenuController.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    private static final String ABORT_OPERATION = "Operation has been canceled. Not found such command.";
+
+    private final MenuView menuView = new MenuView();
+    private final MenuService menuService = new MenuService();
+
+    public void run() {
+        menuService.addMainMenuCommands(this);
+        menuView.update(WELCOME);
+
+        String userInput = askForInput(USER_COMMAND);
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/controller/MenuController.java:24]: Lines 28-31:
This is incorrect. What you want to achieve here is print either `OPERATION_SUCCESS` or `ABORT_OPERATION`.
Ask yourself now @BartoliniAndBorderCollies, what will happen if `executeCommand` return true? And what will happen when the same command returns false?

SPOILER and answer below:
If that method return `false`, app will correctly print only `ABORT_OPERATION`, but when it returns `true` it'll print both the success and abort messages.

File: src/main/java/org/klodnicki/controller/AccountController.java (hunk 4 above)
Comment (JasinskiSz) [reply src/main/java/org/klodnicki/controller/AccountController.java:44]: Those messages are now created in `MenuController`.

**Review by JasinskiSz — CHANGES_REQUESTED** [review]
Great job with `Account` class and overall!

One more thing with logical error in the `MenuController` class. Please check my review comment there.

File: src/main/java/org/klodnicki/service/MenuService.java (hunk 7 above)
Comment (JasinskiSz) [reply src/main/java/org/klodnicki/service/MenuService.java]: Done, TODO no longer in the code.

File: src/main/java/org/klodnicki/entity/Account.java (hunk 6 above)
Comment (JasinskiSz) [reply src/main/java/org/klodnicki/entity/Account.java]: @BartoliniAndBorderCollies great job on this issue! Done.

File: src/main/java/org/klodnicki/controller/MenuController.java (hunk 5 above)
Comment (JasinskiSz) [reply src/main/java/org/klodnicki/controller/MenuController.java]: Done.

**Comment by BartoliniAndBorderCollies** [comment]
OK great! thank you so much for that suggestion! I made changes, everything should be fine right now!

File: src/main/java/org/klodnicki/controller/AccountController.java [hunk 10]
```diff
@@ -14,27 +14,32 @@ public class AccountController {
     private static final String PHONE_NUMBER = "Phone number:";
     private static final String EMAIL = "Email:";
     private static final String ADDRESS = "Address:";
+    private static final String ABORT_OPERATION = "An operation has been canceled.";
+    private static final String SUCCESS = "Success! An account has been created!";
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/controller/AccountController.java]:
```suggestion
    private static final String SUCCESS_ACCOUNT_CREATION = "Success! An account has been created!";
```

File: src/main/java/org/klodnicki/controller/AccountController.java [hunk 11]
```diff
@@ -14,27 +14,32 @@ public class AccountController {
...
 
+        try {
             accountService.create(firstName, secondName, lastName, pesel, phoneNumber, email, address);
-
         } catch (IllegalArgumentException e) {
             menuController.displayOnMenu(e.getMessage());
+            menuController.displayOnMenu(ABORT_OPERATION);
+            return;
         }
+        menuController.displayOnMenu(SUCCESS);
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/controller/AccountController.java]:
```suggestion
        menuController.displayOnMenu(SUCCESS_ACCOUNT_CREATION);
```

**Review by JasinskiSz — DISMISSED** [review]
I'm only leaving a suggestion to change name of one constant.
Other than that, you can go with merge.

File: src/main/java/org/klodnicki/controller/AccountController.java (hunk 11 above)
Comment (BartoliniAndBorderCollies) [reply src/main/java/org/klodnicki/controller/AccountController.java]: OK! done!

**Review by JasinskiSz — APPROVED** [review]
Looking good

**Pull request merged**